### PR TITLE
RATEST-200 E2E Location management workflow test

### PR DESCRIPTION
**Pull request by Ndacyayisenga-droid**
Ticked id RATEST-200
Ticket url https://issues.openmrs.org/browse/RATEST-200

### CODE REVIEW

**Comment by Ndacyayisenga-droid** [comment]
@kdaud @sherrif10 I dont know for what reason I cant access cucumber reports

File: .github/workflows/refapp-2x-location-management.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        run: |
+          docker-compose -f docker/docker-compose-refqa.yml up -d
+      - name: wait for openmrs instance to start
+        run: while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:8080/openmrs/login.htm)" != "200" ]]; do sleep 1; done
+      - name: Run qaframework on selenium
+        run: |
+          echo "cucumber.publish.enabled=true" > qaframework-bdd-tests/src/test/resources/cucumber.properties
+          mvn clean install -DskipTests=true
+          cp -f qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test.properties
+          npm run refapp2LocationManagement
```
Comment (kdaud) [inline .github/workflows/refapp-2x-location-management.yml:38]:
```suggestion
 cd qaframework-bdd-tests
 npm run refapp2LocationManagement
```

**Comment by kdaud** [comment]
> I dont know for what reason I cant access cucumber reports

Its because the command is running against the parent module not within  qaframework-bdd-tests module.

File: .github/workflows/refapp-2x-location-management.yml [hunk 2]
```diff
@@ -35,5 +35,6 @@ jobs:
           echo "cucumber.publish.enabled=true" > qaframework-bdd-tests/src/test/resources/cucumber.properties
           mvn clean install -DskipTests=true
           cp -f qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test-local.properties qaframework-bdd-tests/src/test/resources/org/openmrs/uitestframework/test.properties
-          npm run refapp2LocationManagement
+ cd qaframework-bdd-tests
+ npm run refapp2LocationManagement
```
Comment (kdaud) [inline .github/workflows/refapp-2x-location-management.yml:39]: worth looking into the indention

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/05-location/locations.feature [hunk 3]
```diff
@@ -1 +1,28 @@
...
+  Background:
+    Given a user clicks on the configure metadata link from the home page
+
+
+  @selenium
+  @location
+  Scenario: LOCATION_MANAGEMENT
+    # User story: Add Location Attribute type
+    When the system loads configure metadata link dashboard
+    And a user clicks on Manage Location Attribute Types
```
Comment (mherman22) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/05-location/locations.feature:12]: don't you agree with me that when `a user clicks on Manage Location Attribute Types`, then `the system loads the manage location attribute types page`?

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/05-location/locations.feature [hunk 4]
```diff
@@ -1 +1,28 @@
...
+    And a user clicks on add new location attribute type
```
Comment (mherman22) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/05-location/locations.feature:13]:
```suggestion
    And a user clicks on add new location attribute type button
```

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        quit();
+    }
+
+    @Given("a user clicks on the configure metadata link from the home page")
+    public void clicksConfigureMetaDataLink(){
+        configureMetadataPage = homePage.goToConfigureMetadata();
+    }
+
+    @When("the system loads configure metadata link dashboard")
+    public void loadsConfigureMetaData(){
```
Comment (mherman22) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/LocationManagementSteps.java:65]:
```suggestion
    public void loadsConfigureMetaData() {
```
please do the same for all the other methods

**Review by mherman22 — CHANGES_REQUESTED** [review]

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/05-location/locations.feature [hunk 6]
```diff
@@ -1 +1,28 @@
...
+    When a user clicks on manage location tag
+    Then the user clicks on add new location
+    And a user fills add new location tag form
+    And a user saves add New location tag form
+    Then the New location tag form is saved
+
+    # User story: Manager locations
+    When a user clicks on manager locations
+    Then the user click on add new location
+    And a user fills add new location form
```
Comment (mherman22) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/05-location/locations.feature:28]: what happens when the user fills the add new location form?

**Review by mherman22 — CHANGES_REQUESTED** [review]

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/05-location/locations.feature (hunk 6 above)
Comment (mherman22) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/05-location/locations.feature:28]: @Ndacyayisenga-droid please find time and address these questions

**Comment by sherrif10** [comment]
Hello my dear @Ndacyayisenga-droid  can we give support for this ticket to make you move.

**Pull request closed without merging**